src/Microsoft.Health.ControlPlane.Core/Features/Persistence/IControlPlaneDataStore.cs

```diff
@@ -13,6 +14,10 @@ public interface IControlPlaneDataStore
     {
         Task<IdentityProvider> GetIdentityProviderAsync(string name, CancellationToken cancellationToken);
 
-        Task<IdentityProvider> UpsertIdentityProviderAsync(IdentityProvider identityProvider, CancellationToken cancellationToken);
+        Task<IEnumerable<IdentityProvider>> GetAllIdentityProvidersAsync(CancellationToken cancellationToken);
+
+        Task<UpsertResponse<IdentityProvider>> UpsertIdentityProviderAsync(IdentityProvider identityProvider, string eTag, CancellationToken cancellationToken);
```

Comment:
I think instead of eTag, we should use something more generic like version.

Reply:
The microsoft standards guidelines for apis uses eTag. I plan to leave it as is.